core/server/data/migrations/versions/4.9/06-update-comped-members-status-events.js

```diff
@@ -0,0 +1,36 @@
+const logging = require('@tryghost/logging');
+const {createTransactionalMigration} = require('../../utils.js');
+
+module.exports = createTransactionalMigration(
+    async function up(knex) {
+        logging.info('Updating members_status_events for comped members');
+        const compedMembers = await knex('members')
+            .select('id')
+            .where('status', 'comped');
+
+        if (compedMembers.length === 0) {
+            logging.info('No comped members found - skipping migration');
+            return;
+        } else {
+            logging.info(`Found ${compedMembers.length} comped members - checking members_status_events`);
+        }
+
+        for (const member of compedMembers) {
+            const mostRecentStatusEvent = await knex('members_status_events')
```

Comment:
@allouis @rishabhgrg Is it possible for there to be no `members_status_events` event for a member? 🤔 I don't think so *but* I think we should add a `if (!mostRecentStatusEvent) { logging.warn('huh?'); continue; }` below

Reply:
Yeah i think its pretty unlikely for them to not have one, but no harm in adding the guard i think so sounds good 👍

Reply:
It's not possible for them to not have one, but I'll add the check anyway :relaxed: